core/modules/ccontrol/testCControl.cc

```diff
@@ -686,6 +707,246 @@ static const std::vector<Antlr4CompareQueries> ANTLR4_COMPARE_QUERIES = {
         nullptr,
         "SELECT Source.sourceId,Source.objectId FROM Source WHERE Source.objectId IN(386942193651348) ORDER BY Source.sourceId"
     ),
+
+    // tests the null-safe equals operator
+    Antlr4CompareQueries(
+        "SELECT ra_PS FROM Object WHERE objectId<=>417857368235490",
+        "SELECT ra_PS FROM Object WHERE objectId = 417857368235490",
+        [](query::SelectStmt::Ptr const & selectStatement) {
+            // change the equals op to be the null safe equals op
+            auto whereClauseRef = selectStatement->getWhereClause();
+            auto orTerm = std::dynamic_pointer_cast<query::OrTerm>(whereClauseRef.getRootTerm());
+            auto andTerm = std::dynamic_pointer_cast<query::AndTerm>(orTerm->_terms[0]);
+            auto boolFactor = std::dynamic_pointer_cast<query::BoolFactor>(andTerm->_terms[0]);
+            auto compPredicate = std::dynamic_pointer_cast<query::CompPredicate>(boolFactor->_terms[0]);
+            compPredicate->op = SqlSQL2Tokens::NULL_SAFE_EQUALS_OP;
+        }
+    ),
+
+    // tests the NOT BETWEEN operator
+    Antlr4CompareQueries(
+        "SELECT objectId,ra_PS FROM Object WHERE objectId NOT BETWEEN 417857368235490 AND 420949744686724",
+        "SELECT objectId, ra_PS FROM Object WHERE objectId BETWEEN 417857368235490 AND 420949744686724;",
+        [](query::SelectStmt::Ptr const & selectStatement) {
+            // change the BetweenPredicate's hasNot to true
+            auto whereClauseRef = selectStatement->getWhereClause();
+            auto orTerm = std::dynamic_pointer_cast<query::OrTerm>(whereClauseRef.getRootTerm());
+            auto andTerm = std::dynamic_pointer_cast<query::AndTerm>(orTerm->_terms[0]);
+            auto boolFactor = std::dynamic_pointer_cast<query::BoolFactor>(andTerm->_terms[0]);
+            auto betweenPredicate = std::dynamic_pointer_cast<query::BetweenPredicate>(boolFactor->_terms[0]);
+            betweenPredicate->hasNot = true;
+        }
+    ),
+
+    // tests the && operator.
+    // The Qserv IR converts && to AND as a result of the IR structure and how it serializes it to string.
+    Antlr4CompareQueries(
+        "select objectId, iRadius_SG, ra_PS, decl_PS from Object where iRadius_SG > .5 && ra_PS < 2 && decl_PS < 3;", // &&
```

Comment:
I've found when I come back to these tests later, and when I have to read tests that someone else wrote maybe years earlier, it helps to have come clues about what's going on, what's different, and if that difference was intentional. In this case it may be a little overly verbose, but I'd rather leave more bread crumbs than less.